src/Moq/Mock.cs

```diff
@@ -664,13 +664,12 @@ private static VoidSetupPhrase<T> SetupSetPexProtected<T>(Mock<T> mock, Action<T
 
 					matchers[valueIndex] = new MatchExpression(last.Match);
 
-					if (arguments.Length == 2)
+					for (int i = 0; i < arguments.Length - 1; i++)
```

Comment:
I think it would be clearer to compare `i < valueIndex` to emphasize the fact that we're excluding the `value` parameter.

Reply:
I tried to be consistent with the for loop that is some lines up (line 635).
Let me know if you prefer to (1) leave it as I've done, for maximal consistency between the two code lines AND minimum changes, (2) Fix both for loops, for maximal consistency AND answer your request, or (3) Fix only mine, so at least new code will answer your request (in cost of consistency).

Reply:
Fair enough. Never mind for the moment, let's leave it as you put it.